semgrep/semgrep/util.py

```diff
@@ -64,3 +68,9 @@ def set_flags(debug: bool, quiet: bool) -> None:
     if quiet:
         QUIET = True
         debug_print("QUIET is on")
+
+
+def partition(pred: Callable, iterable: Iterable) -> Tuple[List, List]:
```

Comment:
Yet another reason I need to build that iterables library...

Reply:
I have a good starting point! https://docs.python.org/3.7/library/itertools.html#itertools-recipes ;)

Reply:
IMO there should be a library with roughly this API:

https://www.scala-lang.org/api/current/scala/collection/Iterable.html

Reply:
The closest thing I've seen to this in the Python community is one of these:

- funcy: https://funcy.readthedocs.io/en/stable/
- more-itertools: https://more-itertools.readthedocs.io/en/stable/

It'd be nice to just adopt one of these into the standard lib though, especially since they officially document all these `itertools` recipes. :yuno: just include the functions in the module?

Reply:
oooh funcy looks nice; basically lodash in python

Reply:
let me put my vote in for https://boltons.readthedocs.io/en/latest/

(I use it in semgrep-agent)